profiler/options_test.go

```diff
@@ -16,18 +17,55 @@ import (
 )
 
 func TestOptions(t *testing.T) {
+	t.Run("WithAgentAddr", func(t *testing.T) {
+		var cfg config
+		WithAgentAddr("test:123")(&cfg)
+		expectedURL := "http://test:123/profiling/v1/input"
+		assert.Equal(t, expectedURL, cfg.agentURL)
+		assert.Equal(t, expectedURL, cfg.targetURL())
+	})
+
+	t.Run("WithAgentAddr/override", func(t *testing.T) {
+		os.Setenv("DD_AGENT_HOST", "bad_host")
+		defer os.Unsetenv("DD_AGENT_HOST")
+		os.Setenv("DD_TRACE_AGENT_PORT", "bad_port")
+		defer os.Unsetenv("DD_TRACE_AGENT_PORT")
+		var cfg config
+		WithAgentAddr("test:123")(&cfg)
+		expectedURL := "http://test:123/profiling/v1/input"
+		assert.Equal(t, expectedURL, cfg.agentURL)
+		assert.Equal(t, expectedURL, cfg.targetURL())
+	})
+
 	t.Run("WithAPIKey", func(t *testing.T) {
 		var cfg config
 		WithAPIKey("123")(&cfg)
 		assert.Equal(t, "123", cfg.apiKey)
 	})
 
+	t.Run("WithAPIKey/override", func(t *testing.T) {
+		os.Setenv("DD_API_KEY", "apikey")
+		defer os.Unsetenv("DD_API_KEY")
+		var cfg config
+		WithAPIKey("123")(&cfg)
+		assert.Equal(t, "123", cfg.apiKey)
+	})
+
 	t.Run("WithURL", func(t *testing.T) {
 		var cfg config
 		WithURL("my-url")(&cfg)
 		assert.Equal(t, "my-url", cfg.apiURL)
 	})
 
+	t.Run("WithAPIKey + WithURL", func(t *testing.T) {
```

Comment:
Can you please remove the spaces in the test name? It's been always one word no spaces. They just get replaced with underscores in verbose output which is not that nice.